File: frontend/lib/models/document.dart
class Document {
  final int id;
  final int? dossierId;
  final int? declarationId;
  final String nom;
  final String typeDocument;
  final String? description;
  final String? cheminFichier;
  final String statut;
  final bool actif;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Document({
    required this.id,
    this.dossierId,
    this.declarationId,
    required this.nom,
    required this.typeDocument,
    this.description,
    this.cheminFichier,
    required this.statut,
    required this.actif,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Document.fromJson(Map<String, dynamic> json) {
    return Document(
      id: json['id'] as int,
      dossierId: json['dossier_id'] as int?,
      declarationId: json['declaration_id'] as int?,
      nom: json['nom'] as String,
      typeDocument: json['type_document'] as String,
      description: json['description'] as String?,
      cheminFichier: json['chemin_fichier'] as String?,
      statut: json['statut'] as String? ?? 'Actif',
      actif: json['actif'] as bool? ?? true,
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'dossier_id': dossierId,
      'declaration_id': declarationId,
      'nom': nom,
      'type_document': typeDocument,
      'description': description,
      'chemin_fichier': cheminFichier,
      'statut': statut,
      'actif': actif,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }
}

File: backend/app/models/document.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, ForeignKey, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base


class Document(Base):
    __tablename__ = "documents"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)

    dossier_id: Mapped[int | None] = mapped_column(
        ForeignKey("dossiers.id", ondelete="CASCADE"),
        nullable=True,
        index=True,
    )

    declaration_id: Mapped[int | None] = mapped_column(
        ForeignKey("declarations.id", ondelete="CASCADE"),
        nullable=True,
        index=True,
    )

    nom: Mapped[str] = mapped_column(String(255), nullable=False)

    type_document: Mapped[str] = mapped_column(
        String(100),
        nullable=False,
    )

    description: Mapped[str | None] = mapped_column(
        Text,
        nullable=True,
    )

    chemin_fichier: Mapped[str | None] = mapped_column(
        String(500),
        nullable=True,
    )

    statut: Mapped[str] = mapped_column(
        String(50),
        nullable=False,
        default="Actif",
    )

    actif: Mapped[bool] = mapped_column(
        Boolean,
        nullable=False,
        default=True,
    )

    created_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=datetime.utcnow,
        nullable=False,
    )

    updated_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=datetime.utcnow,
        onupdate=datetime.utcnow,
        nullable=False,
    )

    declaration = relationship(
        "Declaration",
        back_populates="documents",
    )


    dossier = relationship(
        "Dossier",
        back_populates="documents",
    )
